init/cli/logging.go

```diff
@@ -72,13 +82,28 @@ func (f *FileLoggers) SubProcessLogger(name string) launchlib.CreateLogger {
 }
 
 func (f *FileLoggers) OpenFile(path string) (*os.File, error) {
+	if _, ok := f.rotated[path]; !ok {
+		if _, ok := f.flags.(*truncatingFirst); ok {
+			rotate(path)
+			f.rotated[path] = true
+		}
+	}
 	file, err := os.OpenFile(path, f.flags.Get(path), f.mode)
 	if err != nil {
 		return file, errors.Wrapf(err, "could not open logging file '%s'", path)
 	}
 	return file, nil
 }
 
+func rotate(path string) {
+	limit := 5
+	_ = os.Remove(path + "." + strconv.Itoa(limit))
+	for i := limit; i > 0; i-- {
+		_ = os.Rename(path+"."+strconv.Itoa(i-1), path+"."+strconv.Itoa(i))
+	}
+	_ = os.Rename(path, path+".0")
+}
```

Comment:
Alternatively we could rely on mtimes + always increment the index to avoid renaming N files per startup, but that makes the code more complex. Unsure if we should read and log errors if they exist, I suppose we'd have to check if files exist before attempting renames in that case.